lms/djangoapps/completion_api/views.py

```diff
@@ -67,9 +69,16 @@ def get_progress_queryset(self):
         """
         Build a base queryset of relevant StudentProgress objects.
         """
-        objs = StudentProgress.objects.filter(user=self.get_user())
+        objs = StudentProgress.objects.filter(user=self.user)
```

Comment:
@jcdyer Valid point -- I made the change because [`CompletionDetailView.get`]() now calls it 3 times (including the implicit call in `self.get_progress_queryset()`, and I thought that was silly.

I've compromised by changing the it back to a method named `get_user()`, but storing the retrieved value under `self._user`.  cf 3d22449e7e.

Ideally, this would be something done in the upstream code (maybe in middleware?), since the same logic is repeated in several places across the platform, e.g. [Course Blocks API](https://github.com/edx/edx-platform/blob/9c4869c1d59899a1e4a0ed2bf9cf92c77acc4447/lms/djangoapps/course_api/blocks/forms.py#L106-L118), [Grades API](https://github.com/edx/edx-platform/blob/9c4869c1d59899a1e4a0ed2bf9cf92c77acc4447/lms/djangoapps/grades/api/views.py#L62)).